woocommerce/automations
# Add changelog from PRs to readme.txt

# Description
This PR adds the changelog generated as part of the release process into the readme.txt file automatically.

To do this, I have:
- Opened the `readme.txt` file using the GitHub API and converted it from `base64` to `utf-8`.
- Replaced the `== Changelog ==` header with the same header plus the new changelog entry (generated here: https://github.com/woocommerce/automations/blob/5aa3cc7acf304c8f6a240bd2c111eb12a7f1c0b4/lib/utils/changelog/common/changelog.js#L407)
- Converted the new contents of the `readme.txt` file from `utf-8` back to `base64` (required format to update files via the REST api)
- Updated the `readme.txt` file using the GitHub REST API.

# Testing
To test this, you will need to:
- **Fork** the `woocommerce-gutenberg-products-block` and then unfork it (Go to this url: https://support.github.com/contact?tags=rr-forks and type `unfork` into the subject line. Follow the steps from the virtual assistant and wait 5-10 mins for the request to be actioned).
- Enable issues on your new unforked repository (go to `Settings > General > Features > Issues`)
- Activate GH actions
- Give read and write permissions to the workflows: Settings > Actions > General > Workflow permissions
- Allow GH actions to create PRs: Settings > Actions > General
- Create a milestone (I used 9.9.0, but pick whatever you like)
- Clone your repository locally.
- Create a PR on your repository, in this PR you should change the following file: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/.github/workflows/release-pull-request.yml#L27
- Change the highlighted line to read: `uses: opr/automations@add/changelog-entry-to-readme`
- Merge the PR
- Optionally create multiple PRs.
- Assign them all to the milestone you created earlier.
- Create a new branch, `release/9.9.0` or whatever milestone number you picked.
- Push this branch to the remote.
- Go to your repository on GitHub, you should see a new PR created for the release, and it should have added the changelog to `readme.txt` 🙌🏼
